grails-app/views/purchaseOrder/_showOrderItems.gsp

```diff
@@ -132,8 +132,10 @@
                     </g:form>
                 </div>
                 <div id="add-adjustment" class="box">
-                    <g:set var="canManageAdjustments" value="${(order?.status >= OrderStatus.PLACED) && (isApprover
-                            || order?.status == OrderStatus.PENDING)}"/>
+                    <g:set
+                            var="canManageAdjustments"
+                            value="${(order?.status >= OrderStatus.PLACED && isApprover) || (order?.status == OrderStatus.PENDING)}"
```

Comment:
I don't know the context, but it feels like this `order?.status` only works "by a fortune". It seems to be comparing statuses by alphabetic order, not by the `sortOrder` as it was probably meant to be.
Could you double check that?
In my opinion it works by a fortune, but in general it should be written using:
```groovy
OrderStatus.compare(order?.status, OrderStatus.PLACED)
```
Be aware, that by using the `compare` method, a `NullPointerException` might be thrown, because we don't have null safe operators there:
https://github.com/openboxes/openboxes/blob/1e77b282da20c34b75ebec47b896e3528dae095e/src/groovy/org/pih/warehouse/order/OrderStatus.groovy#L33-L35

Reply:
Yes, it was compared by alphabetical order (`>= placed` was returning correct values -> `placed`, `received`, `rejected`). I changed it as you suggested to avoid future problems connected to the order of those statuses.

Reply:
> It seems to be comparing statuses by alphabetic order, not by the sortOrder as it was probably meant to be.

I don't think this is true. Or at least it was not true in Grails 1. 

The only reason it might be true is if within a GSP the order.status is somehow not being coerced into an OrderStatus enum and Grails thinks it's a String, forcing the OrderStatus.PLACED and OrderStatus.PENDING to be compared as Strings.